hubgrep/lib/fetch_results.py

```diff
@@ -74,17 +75,37 @@ def fetch_concurrently(
         future_timeout = (app.config["HOSTING_SERVICE_REQUEST_TIMEOUT"] * 2) + 1
         try:
             for future in futures.as_completed(to_do, timeout=future_timeout):
-                hosting_service_interface, cached_response, _results = future.result()
-                if cached_response.success:
-                    if _results:
-                        _normalize(_results)
-                        results += _results
+                interface_result = future.result()
+                if interface_result.response.success:
+                    if len(interface_result.search_results) > 0:
+                        _normalize(interface_result.search_results)
+                        results += interface_result.search_results
                 else:
-                    failed_responses.append((hosting_service_interface, cached_response))
+                    failed_responses.append(
+                        FailedSearchResult(interface_result.hosting_service_interface, interface_result.response)
+                    )
         except futures._base.TimeoutError as e:
             logger.error("something went wrong with the requests")
             logger.error(e, exc_info=True)
         if failed_responses:
-            logger.warn(f"got some errors: {failed_responses}")
+            logger.warning(f"got some errors: {failed_responses}")
         results = final_sort(keywords, results)
-        return results, failed_responses
+        return AggregatedSearchResults(results, failed_responses)
+
+
+class FailedSearchResult:
+    hosting_service_interface: HostingServiceInterface
+    response: CachedResponse
+
+    def __init__(self, hosting_service_interface: HostingServiceInterface, response: CachedResponse):
+        self.hosting_service_interface = hosting_service_interface
+        self.response = response
+
+
+class AggregatedSearchResults:
+    search_results: List[SearchResult]
+    failed_requests: List[FailedSearchResult]
+
+    def __init__(self, search_results: List[SearchResult], failed_requests: List[FailedSearchResult]):
+        self.search_results = search_results
+        self.failed_requests = failed_requests
```

Comment:
AggregatedSearchResults & FailedSearchResult:
These are then the 2 new classes that outside/frontend stuff will deal with.